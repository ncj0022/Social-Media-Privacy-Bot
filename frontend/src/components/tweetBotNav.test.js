import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { fireEvent } from "@testing-library/react"

import TweetBotNav from './tweetBotNav.component';

let container = null;
let component = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Navbar Testing', () => {

    it('renders correctly on start', () => {
        act(() => {
            render(<TweetBotNav/>, container);
        });
        expect(container).toMatchSnapshot();
    });

    it('Can Switch between Search Functions', () => {
        act(() => {
            component = render(<TweetBotNav/>, container);
        });

        expect(component.state.function).toEqual("Tweet");
        expect(container.querySelector("div div select").value).toEqual("Tweet");

        component.onChangeSearchFunction({
            target: {
                name: "Testing",
                value: "Hashtag"
            }
        });

        expect(component.state.function).toEqual("Hashtag");
        expect(container.querySelector("div div select").value).toEqual("Hashtag")

        component.onChangeSearchFunction({
            target: {
                name: "Testing",
                value: "Account"
            }
        });

        expect(component.state.function).toEqual("Account");
        expect(container.querySelector("div div select").value).toEqual("Account")

        component.onChangeSearchFunction({
            target: {
                name: "Testing",
                value: "Tweet"
            }
        });

        expect(component.state.function).toEqual("Tweet");
        expect(container.querySelector("div div select").value).toEqual("Tweet")
    });

    it('Can handle input changes', () => {
        const mockEvent = {
            target: {
                name: "Testing",
                value: "This is a test"
            }
        }

        act(() => {
            component = render(<TweetBotNav/>, container);
        });

        expect(component.state.input).toEqual("");
        expect(container.querySelector("div div input").value).toEqual("");

        component.onChangeInput(mockEvent);

        expect(component.state.input).toEqual(mockEvent.target.value);
        expect(container.querySelector("div div input").value).toEqual(mockEvent.target.value);
    });
})