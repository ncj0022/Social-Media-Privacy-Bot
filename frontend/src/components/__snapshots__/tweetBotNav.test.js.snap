// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Testing renders correctly on start 1`] = `
<div>
  <div
    class="bg-dark"
  >
    <nav
      class="navbar navbar-dark navbar-expand bg-dark navigation-clean"
    >
      <a
        class="navbar-brand"
        href="#"
      >
        <i>
          PROgrammers
        </i>
      </a>
      <div
        class="container align-middle"
      >
        <input
          class="form-control"
          placeholder="Search..."
          type="text"
          value=""
        />
        <select
          class="custom-select"
        >
          <option
            value="Tweet"
          >
            Search by Tweet
          </option>
          <option
            value="Account"
          >
            Search by Account
          </option>
          <option
            value="Hashtag"
          >
            Search by Hashtag
          </option>
        </select>
        <button
          class="btn btn-primary mx-2"
          data-bss-hover-animate="pulse"
          type="button"
        >
          Search!
        </button>
      </div>
    </nav>
    <div>
      <div>
        <figure
          class="text-center"
        >
          <blockquote
            class="blockquote"
          />
        </figure>
      </div>
    </div>
  </div>
</div>
`;
